### ISSUE TITLE
Change Request: Provide a clean way to custom verify messages for admin use 

### ISSUE THREAD

**Issue by moniuch** [eslint/eslint#16947]
### ESLint version

8.35.0

### What problem do you want to solve?

As an administrator on a project I would like to do what `eslint-plugin-only-warn` does, but in a much cleaner, recognized, supported way, ie. without having to fiddle with the [Linter prototype object](https://github.com/bfanger/eslint-plugin-only-warn/blob/main/src/only-warn.js) + be able to selectively downgrade messages by rule.

### What do you think is the correct solution?

This might not be the most common task one would have to perform with ESLint, but occasionally there is a need to downgrade severity of the messages in order to temporarily mute some problems or gradually introduce certain rules into a project's CI pipelines. The aforementioned plugin has been of great help to me, and I am honestly grateful to its author for finding a way, but when I looked into its [source code](https://github.com/bfanger/eslint-plugin-only-warn/blob/main/src/only-warn.js), it didn't feel like that's the way things should be done. Also, it doesn't allow to selectively downgrade by ruleIds.

I would appreciate if there was a possibility to 'hook' into messages just before they are reported and before the process exit code is determined.

I would be fine doing something like (intended usage follows):

```js
// my-verify.js
module.exports = function (messages) {
  return messages.map(message => {
    return (!message.fatal && message.severity === 2)
      ? {...message, severity: 1 }
      : message
    }
  })
}
```

```sh
eslint . --verify ./my-verify.js
```

On a project where eslint rules had been introduced to IDE only, but not in CI, the process of catching up with CI is painful. This solution would allow me to selectively downgrade the violations rule after rule, putting certain rules into effect ('strict mode') after all violations have been removed in code:

```js
// my-strict-rules.js
module.exports = [ 'eqeqeq', '@some-plugin/no-no-no' ];

// my-verify.js
const myStrictRuleIds = require('my-strict-rules');
module.exports = function(messages){
  return messages.map(message => {
    return (!message.fatal && message.severity === 2 && !myStrictRuleIds.includes(message.ruleId))
      ? { ...message, severity: 1 }
      : message
    }
  })
}
```

### Participation

- [ ] I am willing to submit a pull request for this change.

**Comment by Rec0iL99** [comment]
Hi @moniuch, thanks for the issue. Here's my suggestion, you could create a base config file and then create config files for each of your environments. For local development one `.eslintrc.*` where you can set specific rules and another for your CI.

If your config is in the .js format (`.eslintrc.js`), you could try something like this:

```js
rules: {
  'no-console':
    process.env.NODE_ENV === 'development' ? 'warn' : 'error'
}
```

I hope this is what you are looking for if you want to selectively downgrade the severity of rules.

**Comment by moniuch** [comment]
@Rec0iL99 Thanks for your suggestion. What makes the scenario more complex and tedious is the rules come from presets, like `@plugin/recommended`. So I thought I'd just go to the source code for each preset and paste those rules into my CI config, adjusting their severity individually to `warn`, one after another, as I progress fixing codebase. Unfortunately, I entered another hell of parser configurations, which I still haven't figured out, and my eslint command takes forever.
